homeassistant/components/holiday/calendar.py

```diff
@@ -0,0 +1,120 @@
+"""Holiday Calendar."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from holidays import country_holidays
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_PROVINCE, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Holiday Calendar config entry."""
+    country: str = config_entry.data[CONF_COUNTRY]
+    province: str | None = config_entry.data.get(CONF_PROVINCE)
+
+    async_add_entities(
+        [
+            HolidayCalendarEntity(
+                config_entry.title, country, province, unique_id=config_entry.entry_id
+            )
+        ],
+        True,
+    )
+
+
+class HolidayCalendarEntity(CalendarEntity):
+    """Representation of a Holiday Calendar element."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        name: str,
+        country: str,
+        province: str | None,
+        unique_id: str,
+    ) -> None:
+        """Initialize HolidayCalendarEntity."""
+        self._country = country
+        self._province = province
+        self._location = name
+        self._event: CalendarEvent | None = None
+        self._attr_name = None
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=name,
+        )
+
+    def _default_language(self, hass: HomeAssistant) -> str:
```

Comment:
```suggestion
    def _default_language(self) -> str:
```
It's inside the class so you already have self.hass so no need to pass it.

This would now simply take the default language from the country if exist and not the users language in HA.
So if I run HA in English but want a German calendar I know get it in German even if my language is English. I think we would want it to try to use the HA language as default and fallback to the country's default language.

Reply:
Done, thanks.